packages/cli/fern-definition/validator/src/rules/compatible-ir-versions/compatible-ir-versions.ts

```
@@ -0,0 +1,95 @@
+            }
+        ];
+    }
+    return;
+}
+
+export const CompatibleIrVersionsRule: Rule = {
+    name: "compatible-ir-version",
+    create: async () => {
+        return {
+            generatorsYml: {
+                generatorInvocation: async (invocation) => {
+                    const fdr = createFdrGeneratorsSdkService({ token: undefined });
+                    // TODO: is there a better way to do this? I feel like I can't bring a dep on the cli package
+                    // I also feel like we shouldn't throw or error if we can't get the CLI version, so just returning here.
+                    const cliVersion = process.env.CLI_VERSION;
+                    if (cliVersion == null) {
+                        return [];
+                    }
+
+                    // Pull the CLI release to get the IR version
+                    const cliRelease = await fdr.generators.cli.getCliRelease(cliVersion);
+                    if (!cliRelease.ok) {
+                        return [];
+                    }
+                    const cliIrVersion = cliRelease.body.irVersion;
+
+                    // Pull the generator release to get the IR version
+                    let invocationIrVersion: number;
+                    if (invocation["ir-version"] != null) {
+                        // You've overridden the IR version in the generator invocation, let's clean it up
+                        invocationIrVersion = Number(invocation["ir-version"].replace("v", ""));
+                        if (cliIrVersion >= invocationIrVersion) {
+                            // You've overridden the IR version and you're on a CLI that works with it!
+                            return [];
+                        }
+                    } else {
+                        const generatorEntity = await fdr.generators.getGeneratorByImage({
+                            dockerImage: invocation.name
+                        });
+                        if (!generatorEntity.ok) {
```

nit: i think worth adding a comment here that we won't fail hard on failing to make an HTTP request